processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java

```diff
@@ -298,8 +298,7 @@ private void readAndLoadDataFromSortTempFiles() throws Exception {
         try {
           dataHandler.closeHandler();
         } catch (CarbonDataWriterException e) {
-          LOGGER.error(e);
-          throw new Exception("Problem loading data during compaction: " + e.getMessage());
+          LOGGER.error(e, "Error in close data handler");
```

Comment:
not throwing exception to caller?

Reply:
It‘s better not to throw Exception in finally block, so we just swallow this exception.

Reply:
I think it is better not to hide the exception even in finally block, there are multiple operations inside dataHandler.closeHandler(), let it throw exception to caller then we can know which operation cause the exception.
In line 292, 295, 301, please modify to use `new Exception(msg, e) to include the original exception in the new exception.

Reply:
OK~